apps/api/src/app/workflows-v2/usecases/build-step-issues/build-step-issues.usecase.ts

```diff
@@ -276,10 +280,13 @@ export class BuildStepIssuesUsecase {
   }
 
   @Instrument()
-  private validateSkipField(skipLogic: RulesLogic<AdditionalOperation>): StepIssuesDto {
+  private validateSkipField(variableSchema: JSONSchemaDto, skipLogic: RulesLogic<AdditionalOperation>): StepIssuesDto {
     const issues: StepIssuesDto = {};
+    const { primitives } = parseStepVariables(variableSchema);
```

Comment:
Does maybe using the [buildVariables](https://github.com/novuhq/novu/blob/f9067c548e85d1eaa77d0d05a89773b1d48e1418/apps/api/src/app/workflows-v2/util/build-variables.ts#L11) utility instead help to avoid this full on JSON schema parsing?

Reply:
that function doesn't return all valid variables, but the ones that are provided in the control value (if any)